File: 프로그래머스/level1_정수 제곱근 판별.py
def solution(n):
    if n ** 0.5 - int(n ** 0.5) == 0:
        return (int(n ** 0.5) + 1) ** 2
    return -1

print(solution(3))

# 연산 결과와 1로 나눈 나머지의 값이 0이면, 정수
# 굳이 한번 더 연산해서 뺄셈할 필요 X
# def nextSqure(n):
#     sqrt = n ** (1/2)
#
#     if sqrt % 1 == 0:
#         return (sqrt + 1) ** 2
#     return 'no'

File: 프로그래머스 복습(java)/src/level1/level1_정수_제곱근_판별.java
package level1;

public class level1_정수_제곱근_판별 {
    static class Solution {
        public long solution(long n) {
            if (Math.sqrt(n)==(int)Math.sqrt(n)) {
                return (long) Math.pow(Math.sqrt(n) + 1, 2);
            } else {
                return -1;
            }
        }
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.solution(121));
        System.out.println(solution.solution(3));
    }
}
